### DISCUSSION TITLE
Enable seucrity header like x-content-type-options, strict-transport-security on airflow UI

### DISCUSSION THREAD

**Discussion by sustarun** [apache/airflow#36951] [Q&A]
I have an airflow (v2.6.3) instance on linux server. How can I enable the security headers like x-content-type-options, strict-transport-security on the UI?

**Comment by hanleybrand** [comment]
_important note - I've worked with headers for other web apps, but I haven't done so for airflow specifically_

How are you serving it?  It can be easier to add some headers via the webserver (or ingress using kubernetes), for example check out this article:  
https://webdock.io/en/docs/how-guides/security-guides/how-to-configure-security-headers-in-nginx-and-apache

If on kubernetes, your ingress controller probably has a method for passing config snippets, e.g. the nginx ingress controller supports setting configuration values with [pod annotations](https://docs.nginx.com/nginx-ingress-controller/configuration/ingress-resources/advanced-configuration-with-annotations/), including [passing larger config snippets](https://docs.nginx.com/nginx-ingress-controller/configuration/ingress-resources/advanced-configuration-with-snippets/) as a single block of text.

In terms of airflow specific configuration, check out the webserver & api sections of the [security documentation](https://airflow.apache.org/docs/apache-airflow/2.6.3/security/index.html) -- as far as I know, airflow itself can handle some things like enabling CORS for api calls and setting X-Frame-Options to DENY (set config.webserver.x_frame_enabled to False).

As I mentioned above, I haven't really done anything with headers and airflow specifically so there may be a gotcha -- but my experience is it's more efficient and less confusing to set those headers at the httpd/nginx/ingress level instead of the webapp config/implementation level.

**Reply by sustarun** [reply]
Thanks for your reply.

Not using kubernetes, it is a standalone application on linux server. I have checked airflow configuration but there is no way to specify the headers that I need

I am serving using default airflow webserver (gunicorn)

**Reply by potiuk** [reply]
We are about to maka a better documentation on that and generally speakong what @hanleybrand mentioned is right. Generally speaking airflow provides bare foundation for "secure" web application. But if you want to implement various headers and configure your deployment with a number of other security features (for example rate limiting, or specific headers) that is specific to your expectations - you should put a proxy in front (like ingress in K8S or standalone proxy if you are deploying it in your server) .

That for example includes TLS termination - which could happen in the proxy and you should be able to choose and configure your proxy in the way that you as Deployment Manager need.

**Reply by sivasindhur** [reply]
@sustarun do u have nay solution for this. im also facing the same issue
